### DISCUSSION TITLE
API authorization fail

### DISCUSSION THREAD

**Discussion by msugenius** [YetiForceCompany/YetiForceCRM#17203] [How to] [answered]
**Hi there!**
I ran into problem while trying to use API of YetiForce for integration with my custom webform.

## Expected Behavior
When I requesting /webservice/WebserviceStandard/Users/Login I expected authentication token to be returned. So i want to use other routes after recieving this token.  

## Actual Behavior
After request service Users action Login server returns status 401 and error "Invalid data access". Here is my request params and error trace:

![image](https://github.com/YetiForceCompany/YetiForceCRM/assets/72615150/5be8d6e9-4ba6-41f5-83c9-b039234e1d39)

![image](https://github.com/YetiForceCompany/YetiForceCRM/assets/72615150/50515ae4-2d95-4e70-a891-77f25ce09c4f)

## Trace
============ Error exception ====== 2023-07-18 11:49:24 ======
code: 401 | message: Invalid data access
file: /var/www/yetiforce/api/webservice/WebserviceStandard/Users/Login.php (381)
============ stacktrace: 
#0 /var/www/yetiforce/api/webservice/WebserviceStandard/Users/Login.php(240): Api\WebserviceStandard\Users\Login->checkAccess()
#1 /var/www/yetiforce/api/webservice/Controller.php(125): Api\WebserviceStandard\Users\Login->post()
#2 /var/www/yetiforce/webservice.php(25): Api\Controller->process()
#3 {main}
============ Request 013ab5da ======  2023-07-18 11:49:24  ======
REQUEST_METHOD: POST
REQUEST_URI: /webservice/WebserviceStandard/Users/Login
QUERY_STRING: _container=WebserviceStandard&module=Users&action=Login
PATH_INFO: 
IP: my ip
----------- Headers -----------
x-api-key: JW7qhE56BEfycTAamD9jmGdUrzNAJ85n
x-encrypted: 0
authorization: Basic dGVzdDprODU2WHBRaiE=
user-agent: PostmanRuntime/7.32.3
accept: */*
postman-token: 8216d458-5dda-4b31-8cc1-79462b8a5202
host: domen
accept-encoding: gzip, deflate, br
connection: keep-alive
cookie: YTSID=mfndd6h8910oiq7nc98ofjbj6p
content-length: 0
----------- Request data -----------
Array
(
    [module] => Users
    [action] => Login
)

----------- _GET -----------
Array
(
    [_container] => WebserviceStandard
    [module] => Users
    [action] => Login
)

**So I`ll be so grateful if somebody explain me how to get token and use YetiForce API after that...**

**Comment by rskrzypczak** [comment, marked as answer]
You are missing data for the API user
![image](https://github.com/YetiForceCompany/YetiForceCRM/assets/10221999/1c04e18c-bfaf-4d53-b064-b2b29ef5e75d)

**Reply by msugenius** [reply]
Thank you! But now same problem, only one change - line 398 instead of 381

**Reply by msugenius** [reply]
Here is an error screenshot
![image](https://github.com/YetiForceCompany/YetiForceCRM/assets/72615150/9dd15825-7e7a-41ba-9ae6-4121154e9cf2)

**Reply by msugenius** [reply]
Finally solved, thank you very much!

**Reply by rskrzypczak** [reply]
FYI: the **params** parameter is not required

**Comment by rskrzypczak** [comment]
and it's not a system error so conversion to discussion ;)
